### PR TITLE
Added activation ops + tests

### PR DESCRIPTION
**Pull request by AndreSlavescu**
- hardtanh
- celu
- softsign

### CODE REVIEW

**Comment by geohot** [comment]
Tests should really test multiple values of the params (I know some of the current ones don't, they aren't great either!)

**Comment by AndreSlavescu** [comment]
- refactored activation operators to include type hints 
- tests include more shapes to be tested on

**Comment by geohot** [comment]
Careful with formatting, name:type=val.

And there's no reason to test on different shapes, I mean different alpha and min and max.

**Comment by AndreSlavescu** [comment]
- added params + removed extra shape testing
- fixed formatting

File: test/test_ops.py [hunk 1]
```diff
@@ -139,6 +142,9 @@ def test_exp(self):
     helper_test_op([(45,65)], lambda x: torch.exp(x), Tensor.exp)
   def test_sign(self):
     helper_test_op([(45,65)], lambda x: torch.sign(x), Tensor.sign)
+  def test_softsign(self):
+    for shape in [(45,65), (100, 55), (65, 65)]:
```
Comment (geohot) [inline test/test_ops.py:146]: Shape testing still here

File: test/test_ops.py [hunk 2]
```diff
@@ -209,6 +215,9 @@ def test_log_softmax_other_axis(self):
     helper_test_op([(10,10,10)], lambda x: x.log_softmax(2), lambda x: x.log_softmax(2), atol=1e-7, grad_atol=1e-7)
   def test_tanh(self):
     helper_test_op([(45,65)], lambda x: x.tanh(), Tensor.tanh, atol=1e-6, grad_atol=1e-6)
+  def test_hardtanh(self):
+    for val in range(10, 30):
```
Comment (geohot) [inline test/test_ops.py:219]: 20 runs?

File: tinygrad/tensor.py [hunk 3]
```diff
@@ -375,17 +375,20 @@ def reciprocal(self): return 1.0/self
...
   def hardswish(self): return self * (self+3).relu6() * (1/6)
   def tanh(self): return 2.0 * ((2.0 * self).sigmoid()) - 1.0
+  def hardtanh(self, min_val:int=-1, max_val:int=1): return self.minimum(max_val).maximum(min_val)
   def gelu(self): return 0.5 * self * (1 + (self * 0.7978845608 * (1 + 0.044715 * self * self)).tanh())
   def quick_gelu(self): return self * (self * 1.702).sigmoid()
-  def leakyrelu(self, neg_slope=0.01): return self.relu() - (-neg_slope*self).relu()
+  def leakyrelu(self, neg_slope:float=0.01): return self.relu() - (-neg_slope*self).relu()
   def mish(self): return self * self.softplus().tanh()
-  def softplus(self, beta=1): return (1/beta) * (1 + (self*beta).exp()).log()
+  def softplus(self, beta:int=1): return (1/beta) * (1 + (self*beta).exp()).log()
```
Comment (geohot) [inline tinygrad/tensor.py:390]: Is this really an int?

Comment (AndreSlavescu) [reply tinygrad/tensor.py:390]: seems like pytorch defines beta as an int type parameter:
https://pytorch.org/docs/stable/generated/torch.nn.Softplus.html#torch.nn.Softplus

File: tinygrad/tensor.py [hunk 4]
```diff
@@ -375,17 +375,20 @@ def reciprocal(self): return 1.0/self
...
   def sigmoid(self): return (1.0 + (-self).exp()).reciprocal()
-  def elu(self, alpha=1.0): return self.relu() - alpha*(1-self.exp()).relu()
+  def elu(self, alpha:float=1.0): return self.relu() - alpha*(1-self.exp()).relu()
+  def celu(self, alpha:float=1.0): return self.maximum(0) + (alpha * ((self / alpha).exp() - 1)).minimum(0)
   def swish(self): return self * self.sigmoid()
   def silu(self): return self.swish()   # The SiLU function is also known as the swish function.
   def relu6(self): return self.relu() - (self-6).relu()
   def hardswish(self): return self * (self+3).relu6() * (1/6)
   def tanh(self): return 2.0 * ((2.0 * self).sigmoid()) - 1.0
+  def hardtanh(self, min_val:int=-1, max_val:int=1): return self.minimum(max_val).maximum(min_val)
```
Comment (geohot) [inline tinygrad/tensor.py:385]: Is *this* really an int?

**Pull request merged**